### DISCUSSION TITLE
Sass Scoped Async Component not getting unique css class id's

### DISCUSSION THREAD

**Discussion by lopugit** [quasarframework/quasar#10354] [General - Components / Directives / etc]
I've set up a test repository as I don't know how to get <style> tag based styling on jsfiddle.

I'm having a problem where a recursive async component seems to be getting the same scope/css id across instances.

Is it possible to give unique id's to scoped async components that are nested? So if I spawn a component in itself it can have unique css?

This is a link to the repo, you can look at the files scopedcomp and childcomp if you want.

It's a vue 2 quasar template project

https://github.com/lopugit/tmp

https://user-images.githubusercontent.com/13629190/129189155-c9bbc6c4-390e-403e-81b7-1791f9561f5b.mp4

**Comment by ontwikkelfabriek** [comment]
@lopugit 

Hi, I checked out and ran your code. Here are my comments 

> recursive async component

You do not have a recursive component. You have 2 components of different type that can contain each other. Recursive in this context means that a component of a certain type has direct children and descendants of components of the same type ( like a tree node)

> Is it possible to give unique id's to scoped async components that are nested? 

If you build your recursive component structure from a json config object, you can easily set different id's for each node component.

Here's a stackoverflow link:
https://stackoverflow.com/questions/61991949/render-functions-and-recursive-components-in-vue

See the last section about **unique** id's in recursive components:
https://jeri-omit.medium.com/recursive-components-in-vue-js-1ecc5e3d554d

Also for recursive components usually it is considered a better practice in Vue to use **render** functions instead of **templates** :
https://stackoverflow.com/questions/42170352/vue-js-recursive-call-of-the-component-through-render-function-runtime-only/53651645

**Reply by lopugit** [reply]
Thanks for responding @ontwikkelfabriek 

I don't think you fully understood my problem, I'm trying to change the `data-v-123` value for my async components, as each component gets the same `data-v-123` value on them, shouldn't this value change for each instance of a component? So that scoped css actually scopes to just that instance of the component?

Basically right now
```
scopedcomp.data-v = 123
  -> childcomp.data-v = 456
    -> scopedcomp.data-v = 123
```

Can I get the 3rd scopedcomp.data-v value from scoped css to be different to the first?
```
scopedcomp.data-v = 123
  -> childcomp.data-v = 456
    -> scopedcomp.data-v = 789
```

**Reply by ontwikkelfabriek** [reply]
@lopugit 

> Can I get the 3rd scopedcomp.data-v value from scoped css to be different to the first?

I don't think so.

The css in scoped values apply to ALL instances of a component of the same type. That's why it probably has the same `data-v` value. As far as Vue is concerned it's good enough for applying it's scoped style.

Apart from that I think you should take a step back. You shouldn't  set custom style/css on `data-v` values, that's what id's/classes are for.
